# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: 00d83f1d02e0
# Full Hash: 00d83f1d02e015735d580045524eceeeccdc4e28
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 1 - Allow calling BrowserParent::InitRendering multiple times, and remove RenderFrame dependency from nsFrameLoader. r=kats
#   
#   This cleanup will simplify refactoring nsFrameLoader later.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31429
# ==============================================================================

diff -r bdd76368ddc1 -r 00d83f1d02e0 dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Wed May 22 18:37:34 2019 +0300
+++ b/dom/base/nsFrameLoader.cpp	Wed Apr 24 22:36:27 2019 -0500
@@ -1063,15 +1063,10 @@
       return true;
     }
 
-    RenderFrame* rf =
-        mBrowserParent ? mBrowserParent->GetRenderFrame() : nullptr;
-
-    if (!rf || !rf->AttachLayerManager()) {
-      // This is just not going to work.
+    if (!mBrowserParent->Show(
+            size, ParentWindowIsActive(mOwnerContent->OwnerDoc()))) {
       return false;
     }
-
-    mBrowserParent->Show(size, ParentWindowIsActive(mOwnerContent->OwnerDoc()));
     mRemoteBrowserShown = true;
 
     nsCOMPtr<nsIObserverService> os = services::GetObserverService();